qa/workunits/rbd/rbd-nbd.sh

```diff
@@ -442,4 +444,28 @@ _sudo rbd device detach ${DEV} --device-type nbd
 expect_false get_pid ${POOL}
 DEV=
 
+# test discard granularity with journaling
+rbd feature enable ${POOL}/${IMAGE} journaling
+BS=4096
+rbd config global set client rbd_discard_granularity_bytes ${BS}
+DEV=`_sudo rbd device --device-type nbd map ${POOL}/${IMAGE}`
+get_pid ${POOL}
+_sudo blkdiscard --offset 0 --length $((BS+512)) ${DEV}
+# Wait for commit log to be empty, 10 seconds should be well enough
+TRIES=0
+QUEUE_LENGTH=`rbd journal inspect --pool ${POOL} --image ${IMAGE} --verbose |& awk '/entries inspected/ {print $1}' 2>&1`
+while [ ${TRIES} -lt 10 ] && [ ${QUEUE_LENGTH} -gt 0 ]; do
+  QUEUE_LENGTH=`rbd journal inspect --pool ${POOL} --image ${IMAGE} --verbose |& awk '/entries inspected/ {print $1}' 2>&1`
+  TRIES=$((TRIES+1))
+  sleep 1
+done
+[ ${QUEUE_LENGTH} -eq 0 ]
+unmap_device ${DEV} ${PID}
+rbd feature disable ${POOL}/${IMAGE} journaling
+rbd config global rm client rbd_discard_granularity_bytes
+BS=
+DEV=
```

Comment:
nit: I would keep `DEV` reset as close as possible to `unmap_device`, i.e. where it becomes invalid. Also, I don't see much sense in resetting other variables, which are specific for this test, though I don't mind.